src/eth1spec/rlp.py

```diff
@@ -351,3 +354,29 @@ def decode_item_length(encoded_data: Bytes) -> int:
         )
 
     return 1 + length_length + decoded_data_length
+
+
+#
+# Encoding and decoding dataclasses
+#
+
+
+def encode_dataclasses(raw_dataclass_obj: Any) -> Bytes:
+    """
+    Encoding dataclasses. One should never invoke these methods directly.
+    Rather use the encode function defined above
+    """
+    return encode(astuple(raw_dataclass_obj))
```

Comment:
Interesting! Does this also handle nested dataclasses?

Reply:
In #37, @voith [implemented RLP encoding differently](https://github.com/quilt/eth1.0-specs/pull/37/files#diff-870181b4345e7de97307aa360f3958e009891c4019204a09ce59e8b4f6db0038R351) for block headers.

I don't want to express an opinion either way, just wanted to point out that there are competing styles and we should decide on one.

Reply:
I've been thinking about this a bit, and I'm pretty sure I prefer the explicit style of encoding. For example, something like:

```python
def account_to_tuple(account: Account) -> Tuple[RLP]:
    return (
        account.nonce,
        account.balance,
        root(map_keys(account.storage)),
        crypto.keccak256(account.code),
    )
```

Thoughts?

Reply:
I actually also want the explicit encoding and decoding. I'll try a bit more to add this explicit encoding/decoding feature.